internal/events/aggregator.go

```diff
@@ -206,23 +219,14 @@ func (ag *aggregator) processPins(ctx context.Context, pins []*fftypes.Pin, stat
 		}
 
 		// Extract the message from the batch - where the index is of a topic within a message
-		var msg *fftypes.Message
-		var i int64 = -1
```

Comment:
In the case of `pin.Index == 0` this logic was broken. `msg` and `msgBaseIndex` were incorrect.